open-authoring/OpenAuthoringErrorInterpreter.ts

```diff
@@ -0,0 +1,122 @@
+import OpenAuthoringError from "./OpenAuthoringError";
+import OpenAuthoringContextualError from "./OpenAuthoringContextualError";
+import { Actions } from "./OpenAuthoringModalContainer";
+import { isForkValid } from "./github/api";
+
+export default async function interpretError(error: OpenAuthoringError) : Promise<OpenAuthoringContextualError> {
+    if (!error || !error.code) {
+        return new OpenAuthoringContextualError(
+            false, // should it be presented as a modal? (if not present a toast)
+            "Error", // title
+            "Failed to interpret error. No context provided.", // message (the only thing a toast will present)
+            [{ 
+                message: "Continue",
+                action: Actions.authFlow
+            },
+            { 
+                message: "Cancel",
+                action: Actions.refresh
+            }], // Action buttons
+            false // clear preview cookies
+        )
+    }
+    
+    switch (parseInt(error.code.toString()[0])) {
+        case 4: {
```

Comment:
I don't quite understand the distinction between 4** (client-side) errors are and 5** (server-side) errors here

Reply:
It's half because this keeps the functions lean and more readable and the other half because we may do specific checks based on who's having the error (since 4** is supposed to be cs and 5** ss)